Updates for Terraform policy docs (#2864)
hashicorp/web-unified-docs: content/terraform-policy/v0.1.x (beta)/docs/policy/reference/test/module.mdx

## Configuration model

The `module` block supports the following configuration:

- `module` &nbsp "[`<module_source>`](#module-source)" &nbsp "[`<test_label>`](#test-label)" &nbsp block
  - [`attrs`](#attrs) &nbsp object | optional

>     - `<attr_name>` &nbsp any | repeatable

Reviewer: ```suggestion
    - [`<attr_name>`](#attrs) &nbsp any | repeatable
```
Because the platform automatically creates links for words in backticks in a list, you have to add a destination. 👎

>   - [`expect_failure`](#expect-failure) &nbsp boolean | optional

Reviewer: ```suggestion
  - [`expect_failure`](#expect_failure) &nbsp boolean | optional
```

  - [`inputs`](#inputs) &nbsp block | optional

>     - `<input_name>` &nbsp any | repeatable

Reviewer: ```suggestion
    - [`<input_name>`](#inputs) &nbsp any | repeatable
```

  - [`meta`](#meta) &nbsp object | optional
    - `address` &nbsp string
    - `source` &nbsp string

>     - `version` &nbsp string

Reviewer: ```suggestion
    - [`address`](#meta) &nbsp string
    - [`source`](#meta)` &nbsp string
    - [`version`](#meta) &nbsp string
```

>     - [`source`](#meta)` &nbsp string

Reviewer 2: Stray backtick after the link.

```suggestion
    - [`source`](#meta) &nbsp string
```

## Complete configuration example

The following example demonstrates a module block with all configuration options specified:

<CodeBlockConfig filename="tests/example.policytest.hcl">

```hcl
module "app.terraform.io/my-org/vpc/aws" "all_options" {
  expect_failure = true

  meta = {
    source  = "app.terraform.io/my-org/vpc/aws"
    version = "1.5.0"
    address = "module.vpc"
  }

  attrs = {
    name = "example-vpc"
    cidr = "10.0.0.0/16"
    azs  = ["us-east-1a", "us-east-1b"]
  }

  inputs {
    min_module_version = "2.0.0"
  }
}
```

</CodeBlockConfig>

## Specification

A `module` block supports the following configuration.

### Module source

> The module source (For example: `"app.terraform.io/my-org/vpc/aws"`, `"./modules/networking"`).

Reviewer: ```suggestion
Specifies a URI where Terraform policy retrieves the module from, for example`"app.terraform.io/my-org/vpc/aws"` or `"./modules/networking"`. 
```

## Examples

The following examples demonstrate common mock module configuration patterns for specific use cases.

### Test module version requirements

> In the following example, mock `module` blocks test a policy that enforces minimum module versions — one with a current version and one with an outdated version.

Reviewer: ```suggestion
In the following example, the mock `module` blocks test a policy that enforces minimum module versions. One of the blocks mocks the current version and the second block mocks an outdated version.
```
I think that's what that means?

### Test approved module sources

> In the following example, mock `module` blocks test a policy that validates module sources — one from an approved source and one from an unapproved source.

Reviewer: ```suggestion
In the following example, mock `module` blocks test a policy that validates module sources. One of the blocks mocks an approved source and the second block mocks an unapproved source.
```

### Test version constraints

> In the following example, mock `module` blocks test a policy that enforces version constraints across different module versions—two that pass and one that fails.

Reviewer: ```suggestion
In the following example, mock `module` blocks test a policy that enforces version constraints across different module versions. Two of the mocks are designed to pass and a third is designed to fail.
```
